packages/syft/src/syft/node/node.py

```diff
@@ -748,17 +852,25 @@ def get_role_for_credentials(self, credentials: SyftVerifyKey) -> ServiceRole:
         return role
 
     def handle_api_call(
-        self, api_call: Union[SyftAPICall, SignedSyftAPICall]
+        self,
+        api_call: Union[SyftAPICall, SignedSyftAPICall],
+        job_id: Optional[UID] = None,
+        check_call_location=True,
     ) -> Result[SignedSyftAPICall, Err]:
         # Get the result
-        result = self.handle_api_call_with_unsigned_result(api_call)
+        result = self.handle_api_call_with_unsigned_result(
+            api_call, job_id=job_id, check_call_location=check_call_location
+        )
         # Sign the result
         signed_result = SyftAPIData(data=result).sign(self.signing_key)
 
         return signed_result
 
     def handle_api_call_with_unsigned_result(
-        self, api_call: Union[SyftAPICall, SignedSyftAPICall]
+        self,
+        api_call: Union[SyftAPICall, SignedSyftAPICall],
+        job_id: Optional[UID] = None,
+        check_call_location=True,
```

Comment:
I added `check_call_location` here because when a consumer reads a message the message node_uid does not match the nodes id as it was initially sent to a different node. I was not completely sure how to do this properly as the message is signed